appcues/src/test/java/com/appcues/statemachine/states/BeginningExperienceStateTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.appcues.statemachine.states
+
+import com.appcues.data.model.Experience
+import com.appcues.statemachine.Action.StartExperience
+import com.appcues.statemachine.State
+import com.appcues.statemachine.effects.PresentationEffect
+import com.google.common.truth.Truth.assertThat
+import io.mockk.mockk
+import org.junit.Test
+
+internal class BeginningExperienceStateTest {
+
+    @Test
+    fun `BeginningExperienceState SHOULD be instance of STATE`() {
+        // GIVEN
+        val experience = mockk<Experience>()
+        val state = BeginningExperienceState(experience)
+        // THEN
+        assertThat(state).isInstanceOf(State::class.java)
+        assertThat(state.currentExperience).isEqualTo(experience)
+        assertThat(state.currentStepIndex).isNull()
+    }
+
+    @Test
+    fun `take SHOULD ignore listed actions`() {
```

Comment:
check which actions we are not processing (similar test for all other states test)